enums: init with const value with comp time $if

### Describe the bug

Can't initialice an enum with a const int value

### Reproduction Steps

```v
const enum_value = $if linux { 1 } $else { 2 }

pub enum Test {
	a = enum_value
}

fn main() {
	println(Test.a)
}
```

### Expected Behavior

output:
1 if the build env is linux else 2

### Current Behavior

```v
/tmp/v_1000/break.tmp.c:884: error: '_const_main__enum_value' undeclared
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
make: *** [Makefile:11: run] Error 1
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V full version: V 0.4.8 aa3ab3f.2f85036

### Environment details (OS name and version, etc.)

V full version: V 0.4.8 aa3ab3f.2f85036
OS: linux, Ubuntu 24.04.1 LTS
Processor: 16 cpus, 64bit, little endian, AMD Ryzen 7 7840HS w/ Radeon 780M Graphics

vexe: /home/esquerbatua/git/v/v
vexe mtime: 2024-10-01 19:04:41

vroot: OK, value: /home/esquerbatua/git/v
VMODULES: OK, value: /home/esquerbatua/.vmodules
VTMP: OK, value: /tmp/v_1000

Git version: git version 2.43.0
Git vroot status: 0.4.8-37-g2f850361-dirty
.git/config present: true

CC version: cc (Ubuntu 13.2.0-23ubuntu4) 13.2.0
thirdparty/tcc status: thirdparty-linux-amd64 0134e9b9
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -4406,6 +4406,8 @@ fn (mut g Gen) enum_decl(node ast.EnumDecl) {
 				const_def := g.global_const_defs[util.no_dots(field.expr.name)]
 				if const_def.def.starts_with('#define') {
 					g.enum_typedefs.write_string(const_def.def.all_after_last(' '))
+				} else if const_def.def.contains('const') {
+					g.enum_typedefs.write_string(const_def.def.all_after_last('=').all_before_last(';'))
 				} else {
 					g.enum_typedefs.write_string(expr_str)
 				}

--- a/vlib/v/tests/enums/enum_with_comptime_const_test.v
+++ b/vlib/v/tests/enums/enum_with_comptime_const_test.v
@@ -1,0 +1,10 @@
+const enum_value = $if linux { 1 } $else { 2 }
+
+pub enum Test {
+	a = enum_value
+}
+
+fn test_enum_with_comptime_const() {
+	println(Test.a)
+	assert true
+}

